src/forms/Y2021/irsForms/ScheduleSE.ts

```diff
@@ -1,12 +1,140 @@
-import { TaxPayer } from 'ustaxes/core/data'
+import Form, { FormTag } from 'ustaxes/core/irsForms/Form'
+import F1040 from './F1040'
+import TaxPayer from 'ustaxes/core/data/TaxPayer'
+import { sumRoundedFields } from 'ustaxes/core/irsForms/util'
 
-// TODO
-export default class ScheduleSE {
-  tp: TaxPayer
+export default class ScheduleSE extends Form {
+  tag: FormTag = 'f1040sse'
+  f1040: F1040
+  sequenceIndex = 14
 
-  constructor(tp: TaxPayer) {
-    this.tp = tp
+  constructor(f1040: F1040) {
+    super()
+    this.f1040 = f1040
   }
 
-  l6 = (): number | undefined => undefined
+  postL4Field = (f: () => number | undefined): number | undefined => {
+    if (this.l4c() < 400) {
+      return undefined
+    }
+    return f()
+  }
+
+  l8aRelatedField = (f: () => number | undefined): number | undefined => {
+    return this.postL4Field(() => {
+      if (this.l8a() ?? 0 >= 142800) {
+        return undefined
+      }
+      return f()
+    })
+  }
+
+  l1a = (): number => {
+    const schFL34 = 0 // TODO: Net farm profit or (loss) from Schedule F, line 34
+    const k1B14 = 0 // TODO: If a farm partnership
+    return schFL34 + k1B14
+  }
+
+  l1b = (): number => 0
+
+  l2 = (): number => {
+    const schFL34 = 0 // TODO: Net farm profit or (loss) from Schedule F, line 34
+    const k1B14 = this.f1040.info.scheduleK1Form1065s.reduce(
+      (c, k1) => c + k1.selfEmploymentEarningsA,
+      0
+    )
+    return schFL34 + k1B14
+  }
+
+  l3 = (): number => sumRoundedFields([this.l1a(), this.l1b(), this.l2()])
```

Comment:
It looks like there was a multi-file find-and-replace to introduce rounding at every form field where values are summed together. I understand why you might want this, but I think it will lead to errors.

We try to round fields only one time. [This happens here when the PDFs are filled](https://github.com/ustaxes/UsTaxes/blob/53c48971731439676cc00c6cb2ae6e781fcf2b75/src/core/pdfFiller/fillPdf.ts#L32). So any call to a line method should always produce an unrounded value, and it should only be when the value is computed for display in the PDF that it is rounded.

Suggest this function is not used and the replace is reverted.

It sounded though like you found a case in Schedule D where the rounding was not done properly? Can we figure out why the fillPDF method was not working properly?

Reply:
I actually looked over the instances I changed because I knew it would be dangerous.  The places looked safe to do it.  However, the IRS link you sent makes me question that, but it also makes me question what the IRS really means.  The problem I was running into is that if you add fields prior to rounding, the final math does not work out.  For example, if you have $1.20 and $2.30 and need to sum them, the final form would show $1 + $2 = $4.  So that's what I was fixing here.

Looking at past times I've used the Free File Fillable Forms, I see that on Schedule B, you're forced to round each individual entry before it's added.

I find this line vague:
> If you have to add two or more amounts to figure the amount to enter on a line, include cents when adding the amounts and round off only the total.

What does the IRS consider the "amounts"?  If it says something like line 3 = line 1 + line 2, what is the "amount" of line 1 & 2?  Is it the "displayed amount" or is it the "raw underlying amount".  My guess would be the "displayed amount" because the instructions are likely written for someone manually filling out the forms.  So if the instructions told me to get line X, I'm going to look at what I wrote in line X, which will be the rounded value.

I can revert the changes if you want, but doing so will mean that some forms will have math that looks incorrect.

Also, when I enter in the free file form, I'll see how they handle the rounding.

Reply:
Yea I think the off by one dollar issue is a well understood one, [see here](https://github.com/ustaxes/UsTaxes/pull/955#discussion_r815455375). I just want the value returned by each of those methods to be as accurate as possible and to contain the rounding issue to one single location.

Reply:
It is vague, but reading the instruction, there’s no reason to talk about adding cents if math is to be done with rounded values only. `round(round(x) + y) == round(x) + round(y)` but `round(x+y) != round(x) + round(y)`.

Reply:
Ok.  I reverted that change.

Reply:
On https://www.freefilefillableforms.com/, they round all numbers as you type them.  So if you enter 2 accounts on line 1 with $10.49 interest for each, they change them to $10 and the total is $20, not $21.  Because of that, the code in this project computes a different tax than the IRS version.  I did an experiment of changing `sumFields` to round all fields before adding.  By doing so, I could get the numbers to match in every field.  I wouldn't do it in this branch, but would you mind if I add a setting to do rounding prior to entering fields.  I'd leave the default as what it is now.

Reply:
I think it’s ok. And even if the fillable forms don’t let you input cents, they also don’t force you to use their math do they? Anyway if we are going to change our rounding strategy across the app that will have to be a single PR on it’s own, not part of a big implementation like this one, which is great by the way.

Reply:
As far as I know, https://www.freefilefillableforms.com requires you to use their math, but I can't remember for sure.  Once this gets merged, I'll probably work on a rounding option PR.  No rush on this one though.